### PR TITLE
fix: ethrpc: emit correct bloom filters

### PR DESCRIPTION
**Pull request by magik6k**
## Related Issues
Fixes https://github.com/filecoin-project/lotus/issues/10196

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/types/ethtypes/eth_types.go [hunk 1]
```diff
@@ -155,23 +155,32 @@ type EthBlock struct {
...
+
 func NewEthBlock(hasTransactions bool) EthBlock {
 	b := EthBlock{
 		Sha3Uncles:       EmptyUncleHash, // Sha3Uncles set to a hardcoded value which is used by some clients to determine if has no uncles.
 		StateRoot:        EmptyEthHash,
 		TransactionsRoot: EmptyRootHash, // TransactionsRoot set to a hardcoded value which is used by some clients to determine if has no transactions.
 		ReceiptsRoot:     EmptyEthHash,
 		Difficulty:       EmptyEthInt,
-		LogsBloom:        EmptyEthBloom[:],
+		LogsBloom:        FullEthBloom[:],
```
Comment (magik6k) [inline chain/types/ethtypes/eth_types.go:183]: _Technically_ a full bloom filter never lies, so practically this should be less broken

**Comment by magik6k** [comment]
Fails in test-itest-fevm are unrelated / fixed in https://github.com/filecoin-project/lotus/pull/10193

**Review by shrenujbansal — APPROVED** [review]

File: node/impl/full/eth.go [hunk 2]
```diff
@@ -1912,6 +1907,7 @@ func newEthTxReceipt(ctx context.Context, tx ethtypes.EthTx, lookup *api.MsgLook
 					continue
 				}
 				if entry.Key == ethtypes.EthTopic1 || entry.Key == ethtypes.EthTopic2 || entry.Key == ethtypes.EthTopic3 || entry.Key == ethtypes.EthTopic4 {
```
Comment (raulk) [inline node/impl/full/eth.go:1909]: I assume that Eth blooms don't factor in the position of the topic? If so, this is fine.

**Review by raulk — APPROVED** [review]

File: node/impl/full/eth.go (hunk 2 above)
Comment (magik6k) [reply node/impl/full/eth.go:1909]: Correct, position doesn't matter

**Pull request merged**